File: cdk/backend_stack.py
from aws_cdk import (
    # Duration,
    Stack,
    aws_dynamodb as dynamodb,
    aws_lambda as lambda_
)
from constructs import Construct


class BackendStack(Stack):
    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)
        table = dynamodb.Table(
            self,
            "Table",
            partition_key=dynamodb.Attribute(
                name="id", type=dynamodb.AttributeType.STRING
            ),
            billing_mode=dynamodb.BillingMode.PAY_PER_REQUEST,
        )
        function = lambda_.Function(
            self, 
                                    "ApiHandler",
            code=lambda_.Code.from_asset("../backend/lambda/"),
            handler="handler.handler",
            runtime=lambda_.Runtime.PYTHON_3_10
        )

        table.grant_read_write_data(function)
        function.add_environment("TABLE_NAME", table.table_name)

File: cdk/backendStack.ts
import * as cdk from "aws-cdk-lib";
import * as dynamodb from "aws-cdk-lib/aws-dynamodb";
import * as lambda from "aws-cdk-lib/aws-lambda";
import { Construct } from "constructs";

export class BackendStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const env: string = process.env.DEPLOY_ENV || "dev";

    const table = new dynamodb.TableV2(this, `table-${env}`, {
      partitionKey: { name: "pk", type: dynamodb.AttributeType.STRING },
      billing: dynamodb.Billing.onDemand(),
      removalPolicy: cdk.RemovalPolicy.DESTROY,
      tags: [{ key: "deployEnv", value: env }],
    });

    const apiHandler = new lambda.Function(this, `function-${env}`, {
      code: lambda.Code.fromAsset("../backend/lambda"),
      handler: "handler.handler",
      runtime: lambda.Runtime.PYTHON_3_10,
    });

    table.grantReadWriteData(apiHandler);
    apiHandler.addEnvironment("TABLE_NAME", table.tableName);
  }
}
